### DISCUSSION TITLE
WIX 4 `bal:DisplayInternalUICondition`

### DISCUSSION THREAD

**Discussion by SvetlovA** [wixtoolset/issues#7350] [Questions] [answered]
Hi.

I want to show custom UI on MsiPackage installation. I'm using it like this:
```
<MsiPackage Id="MainPackage" SourceFile="..\file.msi" Vital="yes" Compressed="yes" ForcePerMachine="yes" bal:DisplayInternalUICondition="">
```

The question is, how to use `bal:DisplayInternalUICondition` to show custom UI on Install Upgrade and Repair and not to show it on Uninstall?

Thank you.

**Comment by nirbar** [comment, marked as answer]
The MSI UI condition is evaluated during plan, so you can't use the package action in it.
However you can use, the bundle action
Something like "WixBundleAction >= 6"

The available values are:

enum BOOTSTRAPPER_ACTION
{
    BOOTSTRAPPER_ACTION_UNKNOWN,
    BOOTSTRAPPER_ACTION_HELP,
    BOOTSTRAPPER_ACTION_LAYOUT,
    BOOTSTRAPPER_ACTION_UNSAFE_UNINSTALL,
    BOOTSTRAPPER_ACTION_UNINSTALL,
    BOOTSTRAPPER_ACTION_CACHE,
    BOOTSTRAPPER_ACTION_INSTALL,
    BOOTSTRAPPER_ACTION_MODIFY,
    BOOTSTRAPPER_ACTION_REPAIR,
    BOOTSTRAPPER_ACTION_UPDATE_REPLACE,
    BOOTSTRAPPER_ACTION_UPDATE_REPLACE_EMBEDDED,
};
